src/mapeo-project.js

```diff
@@ -886,25 +913,26 @@ async function deleteAll(dataType) {
  * @param {MapeoProject['$translation']['dataType']} opts.translation
  * @param {MapeoProject['preset']} opts.preset
  * @param {MapeoProject['field']} opts.field
+ * @returns {Promise<String[]>}
  */
-async function deleteTranslations(opts) {
+async function grabTranslationsToDelete(opts) {
+  /** @type {String[]} */
+  const toDelete = []
   const translations = await opts.translation.getMany()
-  await Promise.all(
-    translations.map(async ({ docId, docRef, docRefType }) => {
-      if (docRefType === 'preset' || docRefType === 'field') {
-        let shouldDelete = false
-        try {
-          const toDelete = await opts[docRefType].getByDocId(docRef.docId)
-          shouldDelete = toDelete.deleted
-        } catch (e) {
-          opts.logger.log(`referred ${docRef.docId} is not found`)
-        }
-        if (shouldDelete) {
-          await opts.translation.delete(docId)
-        }
+  translations.forEach(async ({ docRefType, docRef, docId }) => {
+    if (docRefType === 'field' || docRefType === 'preset') {
+      let doc
+      try {
+        doc = await opts[docRefType].getByVersionId(docRef.versionId)
+      } catch (e) {
+        opts.logger.log(`referred ${docRef.versionId} is not found`)
       }
-    })
-  )
+      if (doc) {
+        toDelete.push(docId)
+      }
+    }
+  })
```

Comment:
Sorry for being unclear earlier. We need to await all of these.

```suggestion
  await Promise.all(
    translations.map(async ({ docRefType, docRef, docId }) => {
      if (docRefType === 'field' || docRefType === 'preset') {
        let doc
        try {
          doc = await opts[docRefType].getByVersionId(docRef.versionId)
        } catch (e) {
          opts.logger.log(`referred ${docRef.versionId} is not found`)
        }
        if (doc) {
          toDelete.push(docId)
        }
      }
    })
  )
```

Also, should we make `toDelete` a `Set`? It's possible for multiple translations to refer to the same field/preset, right?

Reply:
yep, right. Changed on the last commit